.github/workflows/R-CMD-check.yaml

```diff
@@ -1,4 +1,8 @@
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
Any reason for only running on push to main? I definitely from time to time appreciate having CI on branches even before they get to the PR stage

Reply:
We were running duplicate builds of everything on PRs because they're both branches and pull requests, so I copied this from one of our other repos. We could drop lines 4-5 (I think?) if we want builds on all pushes to branches

Reply:
> We were running duplicate builds of everything on PRs because they're both branches and pull requests

Could you point me to an example of this?

Reply:
On [your PR](https://github.com/voltrondata-labs/arrowbench/pull/119), if you expand the builds, there are two versions of each, one appended with `(pull_request)`, the other with `(push)`: 

<img width="863" alt="image" src="https://user-images.githubusercontent.com/1693477/205396556-b49726fb-e59c-4058-af5d-e22ae8274c5d.png">

Reply:
IIUC, those are testing two different things though (check out the details of the checkouts). the `push` is the `HEAD` of the branch, the `pull_request` is the merge of `HEAD` of branch + main. For _that specific_ PR they end up being identical code wise (since I rebased the branch on main a few moments ago), but they can and frequently are quite different

## push

```
Checking out the ref
  /usr/local/bin/git checkout --progress --force -B duckdb-ext refs/remotes/origin/duckdb-ext
  Switched to a new branch 'duckdb-ext'
  branch 'duckdb-ext' set up to track 'origin/duckdb-ext'.
/usr/local/bin/git log -1 --format='%H'
'14f9a4c5797ef2529328c050e512e18010737531'
```
https://github.com/voltrondata-labs/arrowbench/actions/runs/3605143011/jobs/6075232144#step:2:455

## pull_request

```
Checking out the ref
  /usr/local/bin/git checkout --progress --force refs/remotes/pull/119/merge
  HEAD is now at 1dfb8c9 Merge 14f9a4c5797ef2529328c050e512e18010737531 into ada600dea8d6383b802be893b9e26b6b6b38792a
/usr/local/bin/git log -1 --format='%H'
'1dfb8c955aeb476fdc7e6dd055db57cacaa3f2b7'
```
https://github.com/voltrondata-labs/arrowbench/actions/runs/3605143166/jobs/6075232423#step:2:459

Reply:
Hm, interesting. Though it seems like `pull_request` is what I'd expect aside from on the default branch. But I can revert.